lib/Provider/AtLoginProvider.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types = 1);
+
+namespace OCA\TwoFactorEmail\Provider;
+
+use OCP\Authentication\TwoFactorAuth\ILoginSetupProvider;
+use OCP\Template;
+use OCP\IUser;
+use OCA\TwoFactorEmail\Service\Email as EmailService;
+use OCP\IConfig as OCCONFIG;
+use OCP\Authentication\TwoFactorAuth\IRegistry;
+
+class AtLoginProvider implements ILoginSetupProvider {
+
+	public function __construct(IUser $myUser, $mySecret, EmailService $EmailService, OCCONFIG $occonfig, IRegistry $registry, Email $provider) {
+		$this->myUser = $myUser;
```

Comment:
Please honor static code analysis errors and thus define the attibutes as in Provider\Email.php, like "/** @var IUser */ \n private $myUser;".